### ISSUE TITLE
[BUG] The require-memo rule should not be triggered when creating JSX at the top level

### ISSUE THREAD

**Issue by lawvs** [arthurgeron/eslint-plugin-react-usememo#45]
## Prerequisites

The require-memo rule is false positive when creating JSX in the top level.

## Minimal repro

Please provide a minimal piece of code or reproducible example that demonstrates the issue.  

```javascript
// Paste your code here

// Array literal should be wrapped in useMemo() or be static when used as a propeslint(@arthurgeron/react-usememo/require-usememo)
const labels = ['data']
const Content = <Text labels={labels} />
```

### Installed packages versions
**ESLint:** 8.57
**Eslint-plugin-react-usememo:** 2.4.0

### Rule and its options

Which rule from the eslint-plugin-react-usememo is being used? What configuration options are set for this rule?

@arthurgeron/react-usememo/require-usememo

### Error log / Stack trace

```
Array literal should be wrapped in useMemo() or be static when used as a propeslint(@arthurgeron/react-usememo/require-usememo)
```

**Comment by arthurgeron** [comment]
Thank you for reporting this issue!
